querybuilder/query_builder.go

```diff
@@ -222,7 +222,7 @@ func (b *QueryBuilder) ToSql(dialect Dialect) (string, error) {
 
 	// Apply custom filters
 	if len(b.filters) > 0 {
-		q = q.Where(b.filters...)
+		q = q.Where(AndGroups(b.filters...))
```

Comment:
Introducing and adding `AndGroups` logic to surround filters by brackets before AND join